ttt-lexical-patterns.lisp

```diff
@@ -9,18 +9,19 @@
 (defparameter *tense* '(past pres cf))
 (defparameter *coordinator* '(and or but because))
 (defparameter *detformer* '(nquan fquan))
+(defparameter *atomsymbols* "\\[?\\{?\(\\w\|\\d\|-\|\/\|:\|\\.\|\\*\|\\[\|\\]\)\+\\}?")
 (defparameter *semtypes*
-  '(("([A-Z]|_)*\\.PRO" . "D")
-    ("\\|([A-Z]|_)*\\|" . "D") ; doesn't work properly. Does work with '\|' instead of '|' in the symbol passed to atom-semtype? I think CL doesn't allow | in symbols.
-    ("([A-Z]|_)*\\.P" . "D => (D => (S => 2))")
-    ("([A-Z]|_)*\\.PS" . "(S => 2) => ((S => 2) => (S => 2))")
-    ("([A-Z]|_)*\\.N" . "D => (S => 2)")
-    ("([A-Z]|_)*-OF\\.N" . "D => (D => (S => 2))")
-    ("([A-Z]|_)*\\.A" . "D => (S => 2)") ; doc gives two possibilities for this
-    ("([A-Z]|_)*\\.V" . "{D | (D => (S => 2))}^n => (D => (S => 2))")
-    ("([A-Z]|_)*\\.D" . "(D => (S => 2))_{n | p} => D")
-    ("([A-Z]|_)*\\.ADV-A" . "(D => (S => 2))_v => (D => (S => 2))_v")
-    ("([A-Z]|_)*\\.(ADV-E|ADV-S|ADV-F)" . "(S => 2) => (S => 2)")
+  '(("{S}\\.PRO" . "D")
```

Comment:
You can make this slightly more robust by performing the `regex-replace-all` right in the `defparameter`.
e.g.
```
(defparameter *semtypes*
  (mapcar 
    #'(lambda (x) (regex-replace-all "{S}" x *atomsymbols*))
    '(("{S}\\.PRO" . "D")
      ...)))
```

That way we can use `*semtypes*` in other places too without having to do the regex replacement again.

Reply:
Other than that, I'm fine with this hack for now.